test/hotspot/jtreg/compiler/cha/StrengthReduceInterfaceCall.java

```diff
@@ -66,6 +70,18 @@ public static void main(String[] args) {
         run(ThreeLevelHierarchyAbstractVsDefault.class);
         run(ThreeLevelDefaultHierarchy.class);
         run(ThreeLevelDefaultHierarchy1.class);
+
+        // Implementation limitation: CHA is not performed by C1 during inlining through MH linkers.
+        if (!jdk.test.whitebox.code.Compiler.isC1Enabled()) {
```

Comment:
Should you check that C2 is **enabled**?

Reply:
> Should you check that C2 is enabled?

The test has `@requires !vm.graal.enabled`. Do you prefer to have it spelled as `@requires vm.compiler2.enabled` instead?

> May it should be in @requires?

Original test cases apply to both C1 and C2. I could introduce a separate test for MH invoker cases, but IMO keeping relevant test logic co-located is preferred compared to avoiding a configuration check at runtime.